crates/db_schema/src/impls/community.rs

```diff
@@ -419,6 +421,29 @@ impl CommunityActions {
       .await?;
     Ok(())
   }
+
+  pub async fn fetch_largest_subscribed_community(
+    pool: &mut DbPool<'_>,
+    person_id: PersonId,
+  ) -> LemmyResult<Option<CommunityId>> {
+    static CACHE: CacheLock<Option<CommunityId>> = LazyLock::new(build_cache);
+    CACHE
+      .try_get_with((), async move {
```

Comment:
You need to pass person_id as cache key here instead of ()

Reply:
Fixed. I'm not sure if just using the API cache_duration duration makes sense, since everything would get kicked out after 1 second.

Reply:
since the heuristic is still valid (just worse) if a smaller community is fetched, setting the cache duration to something like 24h would be fine. the max_size will then be the limiting factor (LRU evictions), which can be pretty large (2*4byte per entry + maybe 8byte for moka internals (couldn't find info in their docs) would mean a cache size of 60k users would be 1MB in size

Reply:
K, I'll add a custom duration for that one.

Reply:
Yes a larger cache size would also be fine for big instances. On smaller instances the size is limited by the number of active users anyway. Also there is probably no need to change cache duration for debug as it will be fast enough either way.